### ISSUE TITLE
Deserialize primitive boolean array

### ISSUE THREAD

**Issue by MartijnDwars** [eclipse-ee4j/yasson#390]
**Describe the bug**
From the specification:

> Implementations MUST support binding of the following basic Java classes and their corresponding primitive types: [...] java.lang.Boolean

and:

> JSON Binding implementations MUST support the binding of Java arrays of all supported Java types from this chapter into/from JSON array structures as defined in Section 5 of RFC 7159. Arrays of primitive types and multi-dimensional arrays MUST be supported.

**To Reproduce**
```
> System.out.println(Arrays.toString(jsonb.fromJson("[true]", Boolean[].class)));
[true]
> System.out.println(Arrays.toString(jsonb.fromJson("[true]", boolean[].class)));
Mar 16, 2020 4:21:04 PM org.eclipse.yasson.internal.Unmarshaller deserializeItem
SEVERE: [Z cannot be cast to [Ljava.lang.Object;
Exception in thread "main" javax.json.bind.JsonbException: Internal error: [Z cannot be cast to [Ljava.lang.Object;
        at org.eclipse.yasson.internal.Unmarshaller.deserializeItem(Unmarshaller.java:72)
        at org.eclipse.yasson.internal.Unmarshaller.deserialize(Unmarshaller.java:52)
        at org.eclipse.yasson.internal.JsonBinding.deserialize(JsonBinding.java:59)
        at org.eclipse.yasson.internal.JsonBinding.fromJson(JsonBinding.java:66)
        at yasson.example.App.main(App.java:48)
Caused by: java.lang.ClassCastException: [Z cannot be cast to [Ljava.lang.Object;
        at org.eclipse.yasson.internal.serializer.ObjectArrayDeserializer.getInstance(ObjectArrayDeserializer.java:50)
        at org.eclipse.yasson.internal.serializer.ObjectArrayDeserializer.getInstance(ObjectArrayDeserializer.java:26)
        at org.eclipse.yasson.internal.serializer.AbstractContainerDeserializer.deserialize(AbstractContainerDeserializer.java:65)
        at org.eclipse.yasson.internal.Unmarshaller.deserializeItem(Unmarshaller.java:66)
        ... 4 more
```

**Expected behavior**
Deserializing `[true]` to a primitive boolean array `boolean[]` should work just like deserializing `[true]` to a reference Boolean array `Boolean[]`.

**System information:**
 - OS: macOS 10.14.6
 - Java Version: 1.8.0 u241 & 11.0.5
 - Yasson Version: 1.0.6

**Comment by Degubi** [comment]
Created a pr for this issue, #392
All other primitive types were working, "boolean" was missing

**Comment by MartijnDwars** [comment]
I'll close this issue since a fix was merged to master and backported to 1.0.x. Thanks!
